lib/webhooks/register.ts

```diff
@@ -252,6 +266,46 @@ function categorizeHandlers(
     toDelete: Object.values(toDelete),
   };
 }
+function areHandlerFieldsEqual(
+  arr1: WebhookHandler,
+  arr2: WebhookHandler,
+): boolean {
+  const includeFieldsEqual = arraysEqual(
+    arr1.includeFields || [],
+    arr2.includeFields || [],
+  );
+  const metafieldNamespacesEqual = arraysEqual(
+    arr1.metafieldNamespaces || [],
+    arr2.metafieldNamespaces || [],
+  );
+  const privateMetafieldNamespacesEqual =
+    arr1.deliveryMethod !== DeliveryMethod.Http ||
+    arr2.deliveryMethod !== DeliveryMethod.Http ||
+    arraysEqual(
+      arr1.privateMetafieldNamespaces || [],
+      arr2.privateMetafieldNamespaces || [],
+    );
+
+  return (
+    includeFieldsEqual &&
+    metafieldNamespacesEqual &&
+    privateMetafieldNamespacesEqual
+  );
+}
+
+function arraysEqual(arr1: any[], arr2: any[]): boolean {
+  if (arr1.length !== arr2.length) {
+    return false;
+  }
+
+  for (let i = 0, len = arr1.length; i < len; i++) {
```

Comment:
nit, I don't think there's any benefit to assign the length to a variable

```suggestion
  for (let i = 0; i < arr1.length; i++) {
```

Reply:
Yeah, for the length of the arrays we're dealing with, re-calculating the length wouldn't really have any performance impact.